### PR TITLE
Refactoring Planning and Control

### PR DESCRIPTION
**Pull request by fangedward**
Tl;dr: Let planning module be responsible for velocity, has some impact on interactions between modules.

This PR consists of a refactor of planning and control. It gives responsibility of producing a velocity to planning, which was originally owned by control. This makes more sense as planning should figure out "what to do", while control should "do it".  In particular, our future planning algorithms (hybrid A*, frenet optimal trajectory) associate a velocity with each waypoint. This PR mainly covers the setup / functionality of `waypoint_planning_operator.py` and `mpc_agent_operator.py`. It moves the `stop_for_agents` call to the planning module from the control module. It also has some lint / flake changes. 

Now the planning module is responsible for sending a `WaypointMessage` which can optionally contain a list of target speeds. This allows `pylot_agent_operator.py` to be unchanged, as it does a bit more speed factor computation in other parts of the code. We can discuss this more and see if we want to migrate all of those features in `pylot_agent_operator.py` into the `waypoint_planner.py`.

**Note** that the interaction between the `waypoint_planner.py` and `mpc_agent.py` has been affected. Since we associate a speed factor with each waypoint, rather compute the speed factor at the time of getting a control, the resulting control is different between the two versions. While this is fine in most cases, it's **particularly different when stopping at a traffic light**. In the current version, MPC produces controls that doesn't stop the car completely before the traffic light. Instead, it rolls to a stop and slightly passes the traffic light. Once it passes the light, the speed factor for lights becomes 1 again as the light no longer applies to the ego vehicle. Then the ego vehicle continues driving along the path.

### CODE REVIEW

File: pylot/planning/waypoint_planning_operator.py [hunk 1]
```diff
@@ -151,12 +181,28 @@ def on_can_bus_update(self, msg, waypoints_stream):
...
             # If waypoints are empty (e.g., reached destination), set waypoint
             # to current vehicle location.
-            self._waypoints = collections.deque([self._vehicle_transform])
+            self._waypoints = deque([self._vehicle_transform])
 
-        head_waypoints = collections.deque(
-            itertools.islice(self._waypoints, 0, DEFAULT_NUM_WAYPOINTS))
+        wp_vector, wp_angle = \
+            pylot.planning.utils.compute_waypoint_vector_and_angle(
+                self._vehicle_transform, self._waypoints, 9)
```
Comment (ICGog) [inline pylot/planning/waypoint_planning_operator.py]: Can you please please use a constant for the waypoint index?

File: pylot/planning/messages.py [hunk 2]
```diff
@@ -2,21 +2,28 @@
 
 
 class WaypointsMessage(erdos.Message):
-    """Message class to be used to send waypoints.
+    """Message class to be used to send waypoints. Optionally can also send
+    a target speed for each waypoint.
```
Comment (ICGog) [inline pylot/planning/messages.py]: Can you please follow the Google comment guidelines https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html ? (i.e., The comment should start with a single line description of the class. You can move the second sentence to a separate line)

File: pylot/planning/rrt_star/rrt_star_planning_operator.py [hunk 3]
```diff
@@ -124,8 +125,9 @@ def on_watermark(self, timestamp, waypoints_stream):
         waypoints = collections.deque(
             itertools.islice(waypoints, 0,
                              DEFAULT_NUM_WAYPOINTS))  # only take 50 meters
-
-        waypoints_stream.send(WaypointsMessage(timestamp, waypoints))
+        target_speeds = deque([MAX_VEL for _ in range(len(waypoints))])
```
Comment (ICGog) [inline pylot/planning/rrt_star/rrt_star_planning_operator.py]: Can you please use self._flags.target_speed instead of MAX_VEL?

File: pylot/planning/utils.py [hunk 4]
```diff
@@ -1,4 +1,9 @@
 import enum
+import math
+from pylot.perception.detection.traffic_light import TrafficLightColor
+
+MAX_VEL = 20  # m/s
```
Comment (ICGog) [inline pylot/planning/utils.py]: Please remove.

File: pylot/planning/waypoint_planning_operator.py [hunk 5]
```diff
@@ -151,12 +181,28 @@ def on_can_bus_update(self, msg, waypoints_stream):
...
+
+        speed_factor, _ = pylot.planning.utils.stop_for_agents(
+            self._vehicle_transform.location, wp_angle, wp_vector,
+            obstacles_msg.obstacles, tl_msg.obstacles, self._flags,
+            self._logger, self._map, timestamp)
 
-        waypoints_stream.send(WaypointsMessage(msg.timestamp, head_waypoints))
+        target_speed = speed_factor * pylot.planning.utils.MAX_VEL
```
Comment (ICGog) [inline pylot/planning/waypoint_planning_operator.py]: Please use target_speed.
